components/Currency/CurrencyAmount/CurrencyAmount.tsx

```diff
@@ -1,24 +1,17 @@
 import React, { FC } from 'react';
 import styled from 'styled-components';
 
-import {
-	formatCurrency,
-	FormatCurrencyOptions,
-	formatNumber,
-	NumericValue,
-	toBigNumber,
-} from 'utils/formatters/number';
-
-import { CurrencyKey } from 'constants/currency';
+import { formatCurrency, FormatCurrencyOptions, formatNumber } from 'utils/formatters/number';
 
 import { ContainerRowMixin } from '../common';
+import { wei, WeiSource } from '@synthetixio/wei';
 
 type CurrencyAmountProps = {
-	currencyKey: CurrencyKey;
-	amount: NumericValue;
-	totalValue: NumericValue;
+	currencyKey: string;
```

Comment:
I would just leave `CurrencyKey` instead of `string` even though it is just a string it avoids problems like you have below where you need to add as `CurrencyKey` because the compiler can't tell they are the same